hbase-protocol-shaded/src/main/protobuf/client/Encryption.proto

```diff
@@ -31,4 +31,5 @@ message WrappedKey {
   required bytes data = 3;
   optional bytes iv = 4;
   optional bytes hash = 5;
+  optional string hash_algorithm = 6 [default = "MD5"];
```

Comment:
Do we have to specify this default at protocol level?

Reply:
In general I think it is a good way to ensure backward compatibility. What do you think? Should I change this part somehow?

Some background info: 

We serialize this protobuf data structure e.g. into the tailer of the encrypted HFiles. Also this travels on the network (e.g. when a client sends a create table request and sending column family descriptors). And we use it when we initialize the crypto contexts when we open a region on a RegionServer.

Adding the `hash_algorithm="MD5"` default value here enabling us to open an encrypted HFile written with an older HBase (where the `hash_algorithm` was not defined) and also enabling the use of an older client (or HBase shell) to e.g create table with encrypted column families when it is connecting to the latest HBase master.

Reply:
Thanks for the explanation @symat, makes sense now.